dataverse-ext-common/src/main/java/edu/harvard/iq/dataverse/importer/metadata/ImporterData.java

```diff
@@ -1,4 +1,66 @@
 package edu.harvard.iq.dataverse.importer.metadata;
 
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class should produce programmatic description of importer form that
+ * will be used for obtain some needed initial data from the user.
+ */
 public class ImporterData {
+
+    private List<ImporterField> importerFormSchema = new ArrayList<>();
+
+    // -------------------- GETTERS --------------------
+
+    public List<ImporterField> getImporterFormSchema() {
+        return importerFormSchema;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public ImporterData addField(ImporterField field) {
+        importerFormSchema.add(field);
+        return this;
+    }
+
+    public ImporterData addDescription(String descriptionKey) {
```

Comment:
I would add javadoc here, is it description regarding whole importer? Where is it going to show? I saw the implementation and I don't understand why are there 2 duplicated keys added?

Reply:
>  and I don't understand why are there 2 duplicated keys added

That's only for have the same text twice, but in different places in the form.